Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/DocumentPropertyEditor.cpp

```diff
@@ -482,8 +476,9 @@ namespace AzToolsFramework
             if (!childWidget)
             {
                 // if there's a null entry in the current place for m_domOrderedChildren,
-                // this entry isn't expanded to that depth and need not follow the change any further
-                AZ_Assert(!IsExpanded(), "m_domOrderedChildren should only contain null entries when collapsed!");
+                // that's ok if this entry isn't expanded to that depth and need not follow the change any further
+                // if we are expanded, then this patch references an unsupported handler, which is probably a problem
+                AZ_Assert(!IsExpanded(), "got patch for unimplemented PropertyHandler");
```

Comment:
I don't think this should assert - attributes on a `PropertyEditor` with no GUI representation can be changed and it should just be a no-op for the UI.